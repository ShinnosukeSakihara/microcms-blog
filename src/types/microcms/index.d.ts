interface InitialColumn {
  createdAt: string
  updatedAt: string
  publishedAt: string
  revisedAt: string
}

export interface Tag extends InitialColumn {
  id: string
  name: string
}

export interface Content extends InitialColumn {
  id: string
  tags: Tag[]
  title: string
  description: string
  body: string
}
