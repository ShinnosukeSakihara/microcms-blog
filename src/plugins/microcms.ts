import axios from 'axios'
import { AxiosResponse } from 'axios'

interface Request {
  draftKey?: string
  limit?: number
  offset?: number
  orders?: string
  q?: string
  fields?: string
  ids?: string
  filters?: string
  depth?: string
}
export interface Response<T> {
  contents: T[]
  totalCount: number
  offset: number
  limit: number
}

const microcms = axios.create({
  baseURL: `${process.env.MICROCMS_URL}/api/v1/`,
  headers: {
    'X-API-KEY': process.env.MICROCMS_X_API_KEY,
  },
})

const getContent = <T>(url: string, params?: Request): Promise<AxiosResponse<T>> => {
  return microcms.get(url, { params })
}

const getContents = <T>(url: string, params?: Request): Promise<AxiosResponse<Response<T>>> => {
  return microcms.get(url, { params })
}

export { microcms, getContent, getContents }
