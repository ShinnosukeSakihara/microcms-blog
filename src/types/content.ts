import { Tag } from './tag'
export interface Content {
  id: string
  tags: Tag[]
  emoji: string
  title: string
  description: string
  body: string
  updatedAt: string
}
