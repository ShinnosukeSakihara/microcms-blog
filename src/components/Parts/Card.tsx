import Text from '@/components/DarkMode/Text'
import dayjs from '@/plugins/dayjs'
import { Content } from '@/types/microcms'
import Link from 'next/link'
import emoji from 'node-emoji/lib/emoji.json'
import React from 'react'

interface Props {
  content: Content
}

const App: React.FC<Props> = (props) => {
  const createdAt = Number(dayjs(props.content.createdAt).format('YYYYMMDD'))
  const emojiArr = Object.entries(emoji).map(([key, emoji]) => ({ key: key, emoji: emoji }))
  const num = createdAt % emojiArr.length
  return (
    <>
      <Link href={`/contents/${props.content.id}`} key={props.content.id}>
        <div className="card h-24 cursor-pointer">
          <div className="card_inner flex">
            <div className="emoji flex flex-row items-center justify-center mr-6 w-24 h-24 bg-white rounded-lg">
              <div className="emoji_inner w-12 h-12 text-5xl">{emojiArr[num].emoji}</div>
            </div>
            <div className="text flex-1">
              <Text>
                <h1 className="break-all text-lg font-bold md:text-xl">{props.content.title}</h1>
              </Text>
              <div className="tag pt-1">
                <div className="tag_inner flex">
                  {props.content.tags.map((tag) => {
                    return (
                      <div className="mr-2 p-0 text-href" key={tag.id}>
                        <Link href={`/tags/${tag.id}`} key={props.content.id}>
                          {`#${tag.name}`}
                        </Link>
                      </div>
                    )
                  })}
                </div>
              </div>
            </div>
          </div>
        </div>
      </Link>
    </>
  )
}
export default App
