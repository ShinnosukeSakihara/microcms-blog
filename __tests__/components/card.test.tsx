import Card from '@/components/Parts/Card'
import { Content } from '@/types/microcms'
import React from 'react'
import renderer from 'react-test-renderer'

describe('Card', (): void => {
  const titleOrName = 'Title Name'
  const at = '2021-01-13T08:51:18.019Z'

  test('tag none', () => {
    const content: Content = {
      id: '1',
      createdAt: at,
      updatedAt: at,
      publishedAt: at,
      revisedAt: at,
      tags: [],
      title: titleOrName,
      description: '',
      body: '',
    }
    const component = renderer.create(<Card content={content} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('tag single', () => {
    const content: Content = {
      id: '1',
      createdAt: at,
      updatedAt: at,
      publishedAt: at,
      revisedAt: at,
      tags: [
        {
          id: '1',
          name: titleOrName,
          createdAt: at,
          updatedAt: at,
          publishedAt: at,
          revisedAt: at,
        },
      ],
      title: titleOrName,
      description: '',
      body: '',
    }
    const component = renderer.create(<Card content={content} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('tag multiple', () => {
    const content: Content = {
      id: '1',
      createdAt: at,
      updatedAt: at,
      publishedAt: at,
      revisedAt: at,
      tags: [
        {
          id: '1',
          name: titleOrName,
          createdAt: at,
          updatedAt: at,
          publishedAt: at,
          revisedAt: at,
        },
        {
          id: '2',
          name: titleOrName,
          createdAt: at,
          updatedAt: at,
          publishedAt: at,
          revisedAt: at,
        },
      ],
      title: titleOrName,
      description: '',
      body: '',
    }
    const component = renderer.create(<Card content={content} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
