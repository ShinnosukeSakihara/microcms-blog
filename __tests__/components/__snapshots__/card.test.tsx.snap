// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tag multiple 1`] = `
<div
  className="card h-24 cursor-pointer"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="card_inner flex"
  >
    <div
      className="emoji flex flex-row items-center justify-center mr-6 w-24 h-24 bg-white rounded-lg"
    >
      <div
        className="emoji_inner w-12 h-12 text-5xl"
      >
        ðŸŽµ
      </div>
    </div>
    <div
      className="text flex-1"
    >
      <span
        className="text-black dark:text-white "
      >
        <h1
          className="break-all text-lg font-bold md:text-xl"
        >
          Title Name
        </h1>
      </span>
      <div
        className="tag pt-1"
      >
        <div
          className="tag_inner flex"
        >
          <div
            className="mr-2 p-0 text-href"
          >
            <a
              href="/tags/1"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              #Title Name
            </a>
          </div>
          <div
            className="mr-2 p-0 text-href"
          >
            <a
              href="/tags/2"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              #Title Name
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Card tag none 1`] = `
<div
  className="card h-24 cursor-pointer"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="card_inner flex"
  >
    <div
      className="emoji flex flex-row items-center justify-center mr-6 w-24 h-24 bg-white rounded-lg"
    >
      <div
        className="emoji_inner w-12 h-12 text-5xl"
      >
        ðŸŽµ
      </div>
    </div>
    <div
      className="text flex-1"
    >
      <span
        className="text-black dark:text-white "
      >
        <h1
          className="break-all text-lg font-bold md:text-xl"
        >
          Title Name
        </h1>
      </span>
      <div
        className="tag pt-1"
      >
        <div
          className="tag_inner flex"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Card tag single 1`] = `
<div
  className="card h-24 cursor-pointer"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="card_inner flex"
  >
    <div
      className="emoji flex flex-row items-center justify-center mr-6 w-24 h-24 bg-white rounded-lg"
    >
      <div
        className="emoji_inner w-12 h-12 text-5xl"
      >
        ðŸŽµ
      </div>
    </div>
    <div
      className="text flex-1"
    >
      <span
        className="text-black dark:text-white "
      >
        <h1
          className="break-all text-lg font-bold md:text-xl"
        >
          Title Name
        </h1>
      </span>
      <div
        className="tag pt-1"
      >
        <div
          className="tag_inner flex"
        >
          <div
            className="mr-2 p-0 text-href"
          >
            <a
              href="/tags/1"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              #Title Name
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
